### PR TITLE
Refactor PR validation pipeline

### PR DESCRIPTION
**Pull request by SparkSnail**

### CODE REVIEW

File: azure-pipelines-pr.yml [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+            backwardCompatibleTestOptions: $(backwardCompatibleTestOptions_Linux_2_4)
+            forwardCompatibleTestOptions: $(forwardCompatibleTestOptions_Linux_2_4)
+        - version: '2.4.1'
+          enableForwardCompatibleTests: false
+          enableBackwardCompatibleTests: false
+          jobOptions:
+          - pool: 'Windows'
+            testOptions: ""
+            backwardCompatibleTestOptions: $(backwardCompatibleTestOptions_Windows_2_4)
+            forwardCompatibleTestOptions: $(forwardCompatibleTestOptions_Windows_2_4)
```
Comment (grazy27) [inline azure-pipelines-pr.yml:147]: We can use foreach here, to reduce duplication: 
https://learn.microsoft.com/en-us/azure/devops/pipelines/process/expressions?view=azure-devops#each-keyword

**Comment by SparkSnail** [comment]
@dotnet-policy-service agree company="Microsoft"

File: azure-pipelines-e2e-tests-template.yml [hunk 2]
```diff
@@ -20,17 +20,24 @@ stages:
       - job: Run_${{ replace(option.pool, ' ', '_') }}
         ${{ if eq(lower(option.pool), 'windows') }}:
           pool:
-            vmImage: 'windows-2022'
+            name: Cosmos2MT-AzureDevOps-AgentPool
```
Comment (wudanzy) [inline azure-pipelines-e2e-tests-template.yml:23]: Can it have a different name? Cosmos2MT is kind of internal name.

File: azure-pipelines-e2e-tests-template.yml [hunk 3]
```diff
@@ -56,19 +63,19 @@ stages:
               sudo apt update
               sudo apt -y install maven
               mvn -version
-
+
```
Comment (wudanzy) [inline azure-pipelines-e2e-tests-template.yml:66]: Remove these whitespaces.

File: azure-pipelines-pr.yml [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+  buildConfiguration: 'Release'
+  _SignType: real
+  _TeamName: DotNetSpark
+  MSBUILDSINGLELOADCONTEXT: 1
+  ArtifactPath: '$(Build.ArtifactStagingDirectory)\Microsoft.Spark.Binaries'
+
+  backwardCompatibleRelease: '2.0.0'
+  forwardCompatibleRelease: '2.0.0'
+
+  backwardCompatibleTestOptions_Windows_2_4: ""
```
Comment (wudanzy) [inline azure-pipelines-pr.yml:19]: Do we need them?

File: azure-pipelines-pr.yml (hunk 1 above)
Comment (SparkSnail) [reply azure-pipelines-pr.yml]: updated.

File: azure-pipelines-e2e-tests-template.yml (hunk 2 above)
Comment (SparkSnail) [reply azure-pipelines-e2e-tests-template.yml:23]: It's the only one available Pool for our team in ManagedOSS project, other pipelines also share the pool, the name can't be changed, or we request to create new agent pool. I think it's OK for the name as the pipeline is only for internal purpose, if customer need to run pipeline based on the yaml file, they need to change to their own pool anyway.

File: azure-pipelines-pr.yml (hunk 4 above)
Comment (SparkSnail) [reply azure-pipelines-pr.yml:19]: These variables bring from original yaml file, they are used if backwardCompatibleTest and forwardCompatibleTest set to true.

File: azure-pipelines-e2e-tests-template.yml (hunk 2 above)
Comment (wudanzy) [reply azure-pipelines-e2e-tests-template.yml:23]: OK

File: azure-pipelines-pr.yml (hunk 4 above)
Comment (wudanzy) [reply azure-pipelines-pr.yml:19]: Do you know where they are used? I didn't find.

Comment (SparkSnail) [reply azure-pipelines-pr.yml:19]: These variables are referenced by generating variable name dynamically according to pool type and spark version, code as below:
```
backwardCompatibleTestOptions: $(backwardCompatibleTestOptions_${{ pool }}_${{ split(version, '.')[0] }}_${{ split(version, '.')[1] }})  
forwardCompatibleTestOptions: $(forwardCompatibleTestOptions_${{ pool }}_${{ split(version, '.')[0] }}_${{ split(version, '.')[1] }})
```
`backwardCompatibleTestOptions` and `forwardCompatibleTestOptions` used in azure-pipelines-e2e-tests-template.yml file.

Comment (wudanzy) [reply azure-pipelines-pr.yml:19]: I see, thanks for explanation!

**Review by wudanzy — APPROVED** [review]

**Comment by grazy27** [comment]
Should we reset branch and add changes as a single commit to streamline the history, or we're using squash merge?

**Review by grazy27 — APPROVED** [review]

**Comment by wudanzy** [comment]
I think we can squash merge.

**Pull request merged**